x-pack/apm-server/main.go

```diff
@@ -250,6 +250,9 @@ func runServerWithProcessors(ctx context.Context, runServer beater.RunServerFunc
 func newProfilingCollector(args beater.ServerParams) (*profiling.ElasticCollector, func(context.Context) error, error) {
 	logger := args.Logger.Named("profiling")
 
+	ctx, stopILM := context.WithCancel(context.Background())
+	profiling.ScheduleILMExecution(ctx, logger.Named("ilm"), args.Config.Profiling)
```

Comment:
Should this be moved down close to the NewCollector call so the goroutine won't be left running if this function returns with an error?

Reply:
I actually left it outside to signal that it's a different route than the collector, and because it uses parts of the configuration from `args` that i'd prefer not to propagate to the collector (the ILM strategy here is temporary™️).

I'll move it below after all err handling, just before the `cleanup` closure.